code/_helpers/unsorted.dm

```diff
@@ -311,26 +311,20 @@ Turf and target are seperate in case you want to teleport some distance from a t
 					break
 
 		//update our pda and id if we have them on our person
-		var/list/searching = GetAllContents(searchDepth = 3)
-		var/search_id = 1
-		var/search_pda = 1
+		var/list/searching = src.GetAllContents()
 
 		for(var/A in searching)
-			if( search_id && istype(A,/obj/item/weapon/card/id) )
+			if(istype(A,/obj/item/weapon/card/id))
```

Comment:
Similarly, why did you get rid of the search_id thing here? It's there to keep this loop from arbitrarily looping over the whole list once it's found it's actual target. Is there some need to make sure that every single ID and PDA on someone with their registered name is updated?